### PR TITLE
Add and fix the clock timer on the submit page when submission limit exhausted

### PR DESCRIPTION
**Pull request by Sanji515**
#### Changes proposed in this pull request:

- Added the clock and fix the implementation to work dynamically

- Added the case when the max submissions exceeded

- Link to live demo: http://pr-179-evalai.surge.sh

**<p align="center">Screenshot 1:</p>**

Screenshot from 2019-07-18 14-54-29

**<p align="center">Screenshot 1:</p>**

Screenshot from 2019-07-18 14-54-59

### CODE REVIEW

**Comment by Sanji515** [comment]
@RishabhJain2018 @lunayach @Shekharrajak @galipremsagar 

Please review

File: src/app/components/challenge/challengesubmit/challengesubmit.component.ts [hunk 1]
```diff
@@ -271,11 +311,15 @@ export class ChallengesubmitComponent implements OnInit {
   fetchRemainingSubmissions(challenge, phase) {
     const API_PATH = this.endpointsService.challengeSubmissionsRemainingURL(challenge);
     const SELF = this;
+    SELF.selectedPhaseSubmissions.showClock = false;
+    SELF.selectedPhaseSubmissions.showSubmissionDetails = false;
+    SELF.selectedPhaseSubmissions.maxExceeded = false;
     this.apiService.getUrl(API_PATH).subscribe(
       data => {
-        let phaseDetails;
+        let phaseDetails, eachPhase;
```
Comment (lunayach) [inline src/app/components/challenge/challengesubmit/challengesubmit.component.ts:326]: `Use data.phase.forEach()`

Comment (Sanji515) [reply src/app/components/challenge/challengesubmit/challengesubmit.component.ts:326]: @lunayach done

**Comment by lunayach** [comment]
Should we include a soft-check so that the host may not enter `Max number of submissions/day` > `Max number of submissions/month` > `Max total submissions`?

File: src/app/components/challenge/challengesubmit/challengesubmit.component.ts [hunk 2]
```diff
@@ -107,14 +107,21 @@ export class ChallengesubmitComponent implements OnInit {
   apiCall: any;
 
   /**
-   * Submissions remaining for the selected phase
+   * Selected phase submission conditions
```
Comment (lunayach) [inline src/app/components/challenge/challengesubmit/challengesubmit.component.ts:110]: For the participant, in the `phases` section, I can see the updated no. of allowed submissions but in the `submit` section, they are unchanged and = `100000`. Is it happening on your side as well?

Comment (Sanji515) [reply src/app/components/challenge/challengesubmit/challengesubmit.component.ts:110]: @lunayach it works fine by my side

Comment (lunayach) [reply src/app/components/challenge/challengesubmit/challengesubmit.component.ts:110]: Yes, it worked, after reverting the latest commit.

File: src/app/components/challenge/challengesubmit/challengesubmit.component.ts [hunk 3]
```diff
@@ -107,14 +107,21 @@ export class ChallengesubmitComponent implements OnInit {
...
    */
   selectedPhaseSubmissions = {
     showSubmissionDetails: false,
-    remainingSubmissions: {},
+    showClock: false,
     maxExceeded: false,
+    remainingSubmissions: {},
     maxExceededMessage: '',
-    message: ''
+    clockMessage: ''
```
Comment (lunayach) [inline src/app/components/challenge/challengesubmit/challengesubmit.component.ts:124]:
```suggestion
    clockMessage: String
```
Also, for the above one.

Comment (lunayach) [reply src/app/components/challenge/challengesubmit/challengesubmit.component.ts:124]: Typecasting won't work for the dictionary but we should typecast as much as we should. I'll suggest you use **interfaces** instead (also in future PR's). https://stackoverflow.com/a/39387930/8681069

**Comment by Sanji515** [comment]
> Should we include a soft-check so that the host may not enter Max number of submissions/day > Max number of submissions/month > Max total submissions

@lunayach Yeah it will be good to have. How about opening a new issue for this and will work on this later?

cc: @RishabhJain2018

**Comment by RishabhJain2018** [comment]
yes, let's incorporate it in the new PR.
@Sanji515 Please open an issue here to keep track of it.
cc: @lunayach

**Review by lunayach — APPROVED** [review]
@RishabhJain2018, it's working on my side. Just that, the no. of remaining submissions update only on the page refresh. Should we also open an issue for it?

**Comment by RishabhJain2018** [comment]
> Just that, the no. of remaining submissions update only on the page refresh. Should we also open an issue for it?

Page refresh or by clicking the phase?

**Comment by lunayach** [comment]
> > Just that, the no. of remaining submissions update only on the page refresh. Should we also open an issue for it?
> 
> Page refresh or by clicking the phase?

Page refresh does it or I have to click the radio buttons assigned to different phases (including the one for which I submitted).

**Comment by RishabhJain2018** [comment]
yes, that is fine for now.

**Pull request merged**